### DISCUSSION TITLE
Iran without filter - suggestion - cooperation ایران بدون فیلتر - پیشنهاد - همیاری

### DISCUSSION THREAD

**Discussion by FreedominIran** [hiddify/Hiddify-Manager#130] [Ideas ایدهها]
## شبکه vpn ایران

دوستانم سلام 
قصد دارم ایده ای رو با شما به اشتراک بزارم که میتونه عبور از محدودیت های فیلترینگ رو بیش از پیش کارا و میسر کنه

حجم اجرایی این ایده میتونه خیلی زیاد باشه و کمک و همیاری همه دوستان رو میطلبه { **به امید ایران آزاد** }

ایده به این صورته که ما یک شبکه میسازیم که متشکل از یک بنچمارک و فروشگاه هست.

از چند دیدگاه بررسی میکنم و مفصلا در موردشون به گپ و گفت میپردازیم:

home ip
-----

اولا اشاره کنم: به جز سایز و رفتار و نوع پاکت ها در شبکه یکی از مهم ترین پارامتری که dpi سرور رو باهاش بلاک میکنه رنج ایپی دیتاسنتره
اتفاقی که توی این شبکه میوفته افراد خارجی میتونن سیستم خودشون رو روشن بزارن و home ip خودشون رو در اختیار شبکه ما بزارند
فقط با اجرای یک دستور اونم با دسترسی کاملا محدود (فقط برای عبور ترافیک)
اما براشون مشکل امنیتی پیش نخواهد اومد؛ چرا چون سرور های پر سرعتی هستند که home ip ها به اون ها تونل شدند و عملا فقط واسط کاربر و سرور vpn هستند هیچ سایت یا نرم افزاری با ip اون ها لود نمیشه و براشون هیچ مشکل امنیتی و حقوقی پیش نمیاد، این شرایط رو برای مشخص کردن رنج ایپی ها به صورت دستی برای dpi به شدت سخت و محدود میکنه

جدای از این بیش از 2500 دیتا سنتر در دنیا vps میفروشند [whtop](https://www.whtop.com/directory/category/vps/os/linux) که کاربران میتونن از اون ها هم استفاده کنند، vps بخرند و در اختیار شبکه بگذارند.

datacenters - cdn - API
-----
کاربران میتونند api دیتاسنتر خودشون رو با شبکه به اشتراک بگذارند و با محدودیت و تنظیمات مشخص اجازه شبکه از منابع اکانت دیتا سنترشون رو بدهند
همینطور کاربران میتونند api cdn خودشون رو چه ایرانی و چه خارجی یا حتی api دیتاسنتر های ایرانی خودشون رو با شبکه به اشتراک بگذارند
مشابه توضیحات قسمت اول تنها ترافیک عبوری از این cdn یا سرور ایرانی فقط به سرور اصلی vpn تونل میشه و سایت یا نرم افزاری با ip این سرورها باز نمیشه، لوگی نخواهند داشت و مشکل قانونی و امنیتی ندارند
نحوه اتصال پرتوکل ها و موقعیت های جغرافیایی و محدودیت های در نظر گرفته شده برای api و هزینه ها همگی توسط سرور مدیریت منیج میشند

shoping
-----
فروش vpn برای امنیت در کل دنیا یه کسب و کاره، ولی با توجه به شرایط بد ایران، انتظار و درخواست من اینه که افراد منابع خودشون رو به رایگان با این شبکه به اشتراک بگذارند ولی در صورتی که امکان یا تمایل به اشتراک گذاری رایگان رو ندارید، سرور ها به صورت زمانی رندوم توسط اپ کلاینت خود شبکه hiddify با کاربران مختلف تست کیفیت میشه و اطلاعاتشون بدون ip فقط با کشور و زمان و up time توی بنچمارک قرار میگیره و اماکان فروش ساعتی با کریپتو های کم فی مثل ترون براش مهیا میشه
کاربر برای اتصال به این شبکه به یک ip یا cdn ایران و یک home ip خارج نیاز داره که هزینه شو ساعتی شارژ و پرداخت میکنه و اون رو متصل میکنه به سرور مجازی شخصی خودش و همیشه با ip ثابتش خودش به اینترنت میره

سرور ها و فروشنده ها رنک میگرن و میتونن گروپ بزنن و مسیر اتصال کاربران اون گروپ هر چند دقیقه به صورت کاملا تصادفی سویچ میشه تا باز هم امکان شناسایی به حداقل برسه

admins
-----
ایپی سرور ها فقط توسط سرور های مادر نگهداری میشه که دسترسیش فقط دست ادمین پروژه و 3 کاربر با بیشترین امتیاز هست به هر دلیل یکی از افراد ping نکنه یه تایمی خودشو فرد دیگه ای جایگزین میشه
به این که بلاک چینی باشه هم فکر کردم ولی اینطوری ممکنه نیرو های ارزشی با صرف هزینه به هد برسن و با کسب دسترسی لیست سرور ها رو در بیارن و محدود کنند
این قسمت با مدلی که گفتم راه میوفته و بدون مشکل تا چند سال آینده ادامه پیدا میکنه و بعدش امید دارم به نسل اینترنت ماهواره ای سویچ کنیم یا دیگه دولتی با این حجم فیلتر و محدودیت نداشته باشیم.

ai
-----
در صورت راه اندزای این شبکه میشه رفتار ترافیک شبکه رو مدل کرد (نه محتوای ترافیک رو ها ! رفتارشو) و اونو به به یک هوش مصنوعی داد و سرور هایی که محدود میشند رو لیبل زد و رویکرد و عملکرد dpi رو کشف کرد و صد چندان شبکه رو غیر قابل تشخیص کرد

ai - oebisfelde
-----
میشه قسمتی رو اضافه کرد که روی سرور ها ورد پرس نصب که و یک سایت بلاگ یا فروشگاهی رندوم روش ران کنه با محتوا و متن و تصویر، عملکرد دقیق dpi رو نمیدونیم ولی اگه بخواد چک کنه روی دامین ترافیک vpn یک سایت واقعی وجود داره

no filter
______
با وجود چنین شبکه ای عملا عمو فیلتر چی به این سادگی ها نمیتونه دسترسی به وب آذاد رو محدود کنه
مگر با چندن کامل اینترنت و بالا آوردن اینترانت ملی که اونم بعید میرسه قابل اجرا باشه :ی

**Comment by hiddify-com** [comment]
ایده خوبی است اما مسائل امنیتی ایجاد میکنه که باید به دقت بررسی بشه مثلا
دانلود تورنت و غیره میتونه دردسر ایجاد کنه
از طرفی پرداخت ریالی مسائل امنیتی دیگری داره
اما با رفع این نگرانی‌ها موضوع خیلی خوبه

**Reply by FreedominIran** [reply]
میشه تورنت رو کلا بسیت
پرداختم فقط با ترون آموزش گسترده میدیم برای خرید و استفاده از ترون

**Reply by FreedominIran** [reply]
اگه مشکل دیگه ای هست مطرح کنید در موردش صحبت کنیم
و بعدش یه to do تعریف کنیم، و کارو پیش ببریم

**Reply by hiddify-com** [reply]
درسته ولی اعتماد کردن کاربران مهمه

شاید راه بهتر استفاده از اون کاربران به عنوان ریلی هست
